spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/cql/CassandraAuthenticatedConnectorSpec.scala

```diff
@@ -4,34 +4,79 @@ import com.datastax.spark.connector.embedded.SparkTemplate._
-      assert(session.getCluster.getMetadata.getClusterName != null)
-    }
-  }
-
-  it should "pick up user and password from SparkConf" in {
-    val conf = defaultConf
-      .set(DefaultAuthConfFactory.UserNameParam.name, "cassandra")
-      .set(DefaultAuthConfFactory.PasswordParam.name, "cassandra")
-
-    // would throw exception if connection unsuccessful
-    val conn2 = CassandraConnector(conf)
-    conn2.withSessionDo { session => }
-  }
+   useCassandraConfig(Seq("cassandra-password-auth.yaml.template"))
+   useSparkConf(defaultConf)
+
+   // Wait for the default user to be created in Cassandra.
+   Thread.sleep(1000)
+
+   val conf = defaultConf
+   conf.set(DefaultAuthConfFactory.UserNameParam.name, "cassandra")
+   conf.set(DefaultAuthConfFactory.PasswordParam.name, "cassandra")
+
+   "A CassandraConnector" should "authenticate with username and password when using native protocol" in {
+      val conn2 = CassandraConnector(conf)
+      conn2.withSessionDo { session =>
+         assert(session !== null)
+         assert(session.isClosed === false)
+         assert(session.getCluster.getMetadata.getClusterName != null)
+      }
+   }
+
+   it should "pick up user and password from SparkConf" in {
+      val conf = defaultConf
+         .set(DefaultAuthConfFactory.UserNameParam.name, "cassandra")
+         .set(DefaultAuthConfFactory.PasswordParam.name, "cassandra")
+
+      // would throw exception if connection unsuccessful
+      val conn2 = CassandraConnector(conf)
+      conn2.withSessionDo { session => }
+   }
+   
+   "A DataFrame" should "allow to write data with valid authentication" in {
+
+      val sqlContext = new SQLContext(sc)
+      import sqlContext.implicits._
+
+      val conf = defaultConf
+         .set(DefaultAuthConfFactory.UserNameParam.name, "cassandra")
+         .set(DefaultAuthConfFactory.PasswordParam.name, "cassandra")
+
+      val conn = CassandraConnector(conf)
+
+      val personDF = sc.parallelize(Seq(
+         ("Andy", 28, "America"),
+         ("Kaushal", 25, "India"),
+         ("Desanth", 27, "India"),
+         ("Mahendra", 26, "Rajasthan")))
+         .toDF("name", "age", "address")
+
+      val options = Map("spark_cassandra_auth_username" -> "cassandra",
+         "spark_cassandra_auth_password" -> "cassandra",
+         "keyspace" -> ks, "table" -> "authtest")
+      createKeyspace(conn.openSession())
+      personDF.createCassandraTable(ks, "authtest", Some(Array("address")), Some(Array("age")))(conn)
+      personDF.write.format("org.apache.spark.sql.cassandra").options(options).save();
+   }
+
+   it should "allow to read data with valid authentication" in {
+
+      val csc = new CassandraSQLContext(sc)
+
+      val options = Map("spark_cassandra_auth_username" -> "cassandra",
+         "spark_cassandra_auth_password" -> "cassandra",
+         "keyspace" -> ks, "table" -> "authtest")
+
+      val personDF = csc.read.format("org.apache.spark.sql.cassandra").options(options).load();
```

Comment:
We don't like to make tests interdependent. Ie if a test "reads" from C* the table should always exist and we can't rely on a previous test to make it. I think in this case it would be fine to just combine this with the previous test. So there is just one "read and write data with valid auth" this way the write portion is validated as well. Wdyt?